src/index.js

```diff
@@ -1,7 +1,114 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import L from 'leaflet';
+ReactDOM.render(	
+  <h1>Hello World!</h1>,	
+ document.getElementById('root')
+) 
 
-ReactDOM.render(
-  <h1>Hello World!</h1>,
-  document.getElementById('root')
-)
\ No newline at end of file
+var fs = require('fs');
+var http = require('http');
+var mime = require('mime-types');
+
+var port = process.env.PORT || 5000;
+http.createServer(function (request, response) {
+  let contentType = 'text/plain';
+  let data;
+  let path = request.url;
+
+  if (path === '/') {
+    file = 'index.html';
+  }
+
+  else if (path.indexOf('.') === -1) {
+    file = 'index.html';
+  }
+
+  else {
+    file = '.' + decodeURIComponent(request.url);
+  }
+
+  try {
+    if (file) {
+      console.log('Serving ' + file);
+      data = fs.readFileSync(file);
+      contentType = mime.lookup(file);
+    }
+  } catch (error) {
+    console.log(error);
+
+    data = "Error: " + error.toString();
+    response.statusCode = 404;
+  }
+
+  response.setHeader('Content-Type', contentType + '; charset=utf-8');
+  response.write(data);
+  response.end();
+}).listen(port);
+
+console.log("Listening on port " + port);
+
+class Map extends React.Component {
+  componentDidMount() {
+    // create map
+    this.map = L.map('map', {
+      center: [49.8419, 24.0315],
+      zoom: 16,
+      layers: [
+        L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
+          attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+        }),
+      ]
+    });
+  }
+
+  render() {
+    return <div id="map"></div>
+  }
+}
+
+export default Map;
+
+componentDidMount() {
```

Comment:
everything from line 71 through line 113 is causing your program to not compile successfully. This is because you are calling React-specific functions outside of the class component. Move these functions in between the componentDidMount function and the render function in your Map class. In addition, you have numerous componentDidMount functions and componentDidUpdate functions in your code. Since you have already defined that function on line 52, you should delete the duplicate functions outside of your class and move their contents into one componentDidMount function and one componentDidUpdate function.